credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,317 @@
+   * @return A [VerifiableCredential] instance.
+   *
+   * Example:
+   * ```
+   * val statusListCredential = StatusListCredential.create("http://example.com/statuslistcred/id123", "http://example.com/issuers/1", StatusPurpose.REVOCATION, listOf(vc1,vc2))
+   * ```
+   */
+  @Throws(RuntimeException::class)
+  public fun create(
+    statusListCredentialId: String,
+    issuer: String,
+    statusPurpose: StatusPurpose,
+    issuedCredentials: List<VerifiableCredential>
+  ): VerifiableCredential {
+    val statusListIndexes: List<String>
+    val bitString: String
+
+    try {
+      statusListIndexes = prepareCredentialsForStatusList(statusPurpose, issuedCredentials)
+      bitString = bitstringGeneration(statusListIndexes)
+    } catch (e: Exception) {
+      throw RuntimeException("An error occurred during the creation of the status list credential: ${e.message}", e)
+    }
+
+    val claims = mapOf(STATUS_PURPOSE to statusPurpose.toString().lowercase(), ENCODED_LIST to bitString)
+    val credSubject = CredentialSubject.builder()
+      .id(URI.create(statusListCredentialId))
+      .type("StatusList2021")
+      .claims(claims)
+      .build()
+
+    val vcDataModel = VcDataModel.builder()
+      .id(URI.create(statusListCredentialId))
+      .issuer(URI.create(issuer))
+      .issuanceDate(Date())
+      .context(URI.create("https://w3id.org/vc/status-list/2021/v1"))
+      .type("StatusList2021Credential")
+      .credentialSubject(credSubject)
+      .build()
+
+    return VerifiableCredential(vcDataModel)
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a [VerifiableCredential].
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param statusListCredential The [VerifiableCredential] representing the status list.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the `statusListCredential`.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck, statusListCred)
+   * ```
+   */
+  public fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredential,
+    statusListCredential: VerifiableCredential
+  ): Boolean {
+    val statusListEntryValue: StatusList2021Entry =
+      StatusList2021Entry.fromJsonObject(credentialToValidate.vcDataModel.credentialStatus.jsonObject)
+
+    val statusListCredStatusPurpose: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[STATUS_PURPOSE] as? String?
+
+    if (statusListEntryValue.statusPurpose == null) {
+      throw IllegalArgumentException("Status purpose in the credential to validate is null")
+    }
```

Comment:
Here and elsewhere, you can simplify as shown below
```suggestion
    require (statusListEntryValue.statusPurpose != null) {
      "Status purpose in the credential to validate is null"
    }
```

Reply:
yup sounds good, I think I had this but there was chat to make it more explicit, I'll put it back to require though as I think its better syntax